rasa/core/processor.py

```diff
@@ -666,30 +643,40 @@ def is_action_limit_reached(
             and should_predict_another_action
         )
 
-    async def _predict_and_execute_next_action(
-        self, output_channel: OutputChannel, tracker: DialogueStateTracker
-    ) -> None:
+    async def _run_prediction_loop(
+        self,
+        output_channel: OutputChannel,
+        tracker: DialogueStateTracker,
+        message: Optional[UserMessage] = None,
+    ) -> DialogueStateTracker:
         # keep taking actions decided by the policy until it chooses to 'listen'
         should_predict_another_action = True
-
         # action loop. predicts actions until we hit action listen
         while should_predict_another_action and self._should_handle_message(tracker):
             # this actually just calls the policy's method by the same name
             try:
-                action, prediction = self.predict_next_action(tracker)
+                tracker, action, prediction = self.predict_next_with_tracker_if_should(
+                    tracker, message
+                )
+                message = None
             except ActionLimitReached:
                 logger.warning(
                     "Circuit breaker tripped. Stopped predicting "
                     f"more actions for sender '{tracker.sender_id}'."
                 )
                 if self.on_circuit_break:
                     # call a registered callback
-                    self.on_circuit_break(tracker, output_channel, self.nlg)
+                    self.on_circuit_break(tracker, message.output_channel, self.nlg)
                 break
 
-            should_predict_another_action = await self._run_action(
-                action, tracker, output_channel, self.nlg, prediction
-            )
+            if action:
+                should_predict_another_action = await self._run_action(
+                    action, tracker, output_channel, self.nlg, prediction
+                )
+            else:
+                break
```

Comment:
this case can't happen, right?

Reply:
If it can happen for NLU-only models then I think we should be more vocal about it

Reply:
I think it's expected behaviour to be able to `handle_message` with an NLU only model. It just adds it to the tracker.